### PR TITLE
services/horizon: fix reingest duplicate errors on same ledger range

### LINKED ISSUE #4110: Horizon db reingest command has duplicate error when run multiple times with same ledger range
### What version are you using?
v2.11.0

### What did you do?
I ran this ingest command on an empty db
```horizon db reingest range 962300 963300 --db-url <any_url_here> --history-archive-urls https://history.stellar.org/prd/core-testnet/core_testnet_001,https://history.stellar.org/prd/core-testnet/core_testnet_002 --network-passphrase "Test SDF Network ; September 2015"```

then I ran the same command a second time against the same db, noticed reingest reported errors on the second time, stopped re-ingesting. 

### What did you expect to see?
the reingestion completes to final ledger without errors on console

### What did you see instead?
```
ERRO[2021-11-30T12:22:56.255-08:00] Error in ingestion state machine
current_state="reingestHistoryRange(fromLedger=962300, toLedger=963300, force=false)" 
error="error processing ledger sequence=962976: Error commiting changes from processor: error in 
*processors.LiquidityPoolsTransactionProcessor.Commit: could not flush transaction liquidity 
pools to db: error adding values while inserting to history_transaction_liquidity_pools: 
exec failed: pq: duplicate key value violates unique 
constraint \"index_history_transaction_liquidity_pools_on_ids\"" next_state=stop pid=20551 service=ingest

INFO[2021-11-30T12:22:56.255-08:00] Shut down pid=20551 service=ingest
error processing ledger sequence=962976: Error commiting changes from processor: error in
*processors.LiquidityPoolsTransactionProcessor.Commit: could not flush 
transaction liquidity pools to db: error adding values while inserting to 
history_transaction_liquidity_pools: exec failed: pq: duplicate key value violates unique constraint
"index_history_transaction_liquidity_pools_on_ids"

```

### PR DESCRIPTION
**Pull request by sreuland**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

include liquidity pool tables in the ingest state machine's history operation purge for a given ledger range

### Why

fix error during reingestion due to duplicate errors in liquidity pool table state

closes #4110 

### Known limitations

N/A

### CODE REVIEW

**Review by tamirms — APPROVED** [review]

**Comment by tamirms** [comment]
@sreuland can you rebase on top of release-horizon-v2.12.0 and discard the commits which are not relevant?

**Comment by sreuland** [comment]
> @sreuland can you rebase on top of release-horizon-v2.12.0 and discard the commits which are not relevant?

@tamirms , how about I squash merge this down to one commit to release-horizon-v2.12.0, then I can port that commit to master afterwards?

File: services/horizon/internal/integration/db_test.go [hunk 1]
```diff
@@ -103,8 +209,8 @@ func TestReingestDB(t *testing.T) {
 		fmt.Sprintf("%d", toLedger),
 	))
 
-	// Reingest into the DB
 	tt.NoError(horizoncmd.RootCmd.Execute())
+	tt.NoError(horizoncmd.RootCmd.Execute(), "Repeat the same reingest range against db, should not have errors.")
```
Comment (sreuland) [inline services/horizon/internal/integration/db_test.go:213]: this is the crux of change, verifies repeated ingest works over same range on same db.

**Comment by sreuland** [comment]
> > @sreuland can you rebase on top of release-horizon-v2.12.0 and discard the commits which are not relevant?
> 
> @tamirms , how about I squash merge this down to one commit to release-horizon-v2.12.0, then I can port that commit to master afterwards?

@tamirms , I rebased to the release branch and squashed, should be finished in CI soon, thanks!

**Comment by tamirms** [comment]
great, thanks! I will merge

**Pull request merged**